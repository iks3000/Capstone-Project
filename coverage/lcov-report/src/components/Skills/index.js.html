
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/components/Skills/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> / <a href="index.html">src/components/Skills</a> index.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">73.68% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>14/19</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">64.29% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>9/14</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">57.14% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>4/7</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">77.78% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>14/18</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a>
<a name='L90'></a><a href='#L90'>90</a>
<a name='L91'></a><a href='#L91'>91</a>
<a name='L92'></a><a href='#L92'>92</a>
<a name='L93'></a><a href='#L93'>93</a>
<a name='L94'></a><a href='#L94'>94</a>
<a name='L95'></a><a href='#L95'>95</a>
<a name='L96'></a><a href='#L96'>96</a>
<a name='L97'></a><a href='#L97'>97</a>
<a name='L98'></a><a href='#L98'>98</a>
<a name='L99'></a><a href='#L99'>99</a>
<a name='L100'></a><a href='#L100'>100</a>
<a name='L101'></a><a href='#L101'>101</a>
<a name='L102'></a><a href='#L102'>102</a>
<a name='L103'></a><a href='#L103'>103</a>
<a name='L104'></a><a href='#L104'>104</a>
<a name='L105'></a><a href='#L105'>105</a>
<a name='L106'></a><a href='#L106'>106</a>
<a name='L107'></a><a href='#L107'>107</a>
<a name='L108'></a><a href='#L108'>108</a>
<a name='L109'></a><a href='#L109'>109</a>
<a name='L110'></a><a href='#L110'>110</a>
<a name='L111'></a><a href='#L111'>111</a>
<a name='L112'></a><a href='#L112'>112</a>
<a name='L113'></a><a href='#L113'>113</a>
<a name='L114'></a><a href='#L114'>114</a>
<a name='L115'></a><a href='#L115'>115</a>
<a name='L116'></a><a href='#L116'>116</a>
<a name='L117'></a><a href='#L117'>117</a>
<a name='L118'></a><a href='#L118'>118</a>
<a name='L119'></a><a href='#L119'>119</a>
<a name='L120'></a><a href='#L120'>120</a>
<a name='L121'></a><a href='#L121'>121</a>
<a name='L122'></a><a href='#L122'>122</a>
<a name='L123'></a><a href='#L123'>123</a>
<a name='L124'></a><a href='#L124'>124</a>
<a name='L125'></a><a href='#L125'>125</a>
<a name='L126'></a><a href='#L126'>126</a>
<a name='L127'></a><a href='#L127'>127</a>
<a name='L128'></a><a href='#L128'>128</a>
<a name='L129'></a><a href='#L129'>129</a>
<a name='L130'></a><a href='#L130'>130</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { useEffect, useState } from "react";
import Button from "../Button";
import { useFormik } from "formik";
import * as Yup from "yup";
import { useDispatch } from "react-redux";
import { postSkills } from "../../app/features/skills/skillsSlice";
import PenLine from '../../assets/images/icons/penLine.svg';
import "./style.scss";
&nbsp;
&nbsp;
const Skills = () =&gt; {
  const [isOpen, setIsOpen] = useState(false);
  const [skills, setSkills] = useState(() =&gt; {
    const savedSkills = localStorage.getItem("skills");
    const initialValue = JSON.parse(savedSkills);
    return initialValue || [];
  });
&nbsp;
  const dispatch = useDispatch();
&nbsp;
  //validation schema for skill name and range
  const SkillsSchema = Yup.object().shape({
    title: Yup.string().required("Skill title is a required field"),
    range: Yup.number()
      .min(10, "Skill range must be greater than or equal to 10")
      .max(100, "Skill range must be less than or equal to 100")
      .required("Skill range is a required field"),
  });
&nbsp;
  const formik = useFormik({
    initialValues: {
      title: "",
      range: "",
    },
    validationSchema: SkillsSchema,
    onSubmit: <span class="fstat-no" title="function not covered" >(v</span>alues, { resetForm }) =&gt; {
<span class="cstat-no" title="statement not covered" >      setSkills(<span class="fstat-no" title="function not covered" >(p</span>rev) =&gt; <span class="cstat-no" title="statement not covered" >[...prev, values])</span>;</span>
<span class="cstat-no" title="statement not covered" >      dispatch(postSkills(values));</span>
<span class="cstat-no" title="statement not covered" >      resetForm();</span>
    },
  });
&nbsp;
  useEffect(() =&gt; {
    //saving skills to local storage
    localStorage.setItem("skills", JSON.stringify(skills));
  }, [skills]);
&nbsp;
  const handleOpenEdit = () =&gt; {
    setIsOpen(!isOpen);
  };
&nbsp;
  return (
    &lt;div className="skills" id="skillsComponent"&gt;
&nbsp;
      &lt;div className="skills__header33"&gt;
        &lt;h2 className="skills__title"&gt;Skills&lt;/h2&gt;
        &lt;Button
          icon={&lt;img className="skills__button-ico" src={PenLine} alt="Pen with Line" /&gt;}
          text="Open edit"
          onClick={handleOpenEdit}
        /&gt;
      &lt;/div&gt;
      {isOpen &amp;&amp; (
        &lt;form className="skills__form" onSubmit={formik.handleSubmit}&gt;
          &lt;div className="skills__inputs"&gt;
            &lt;label htmlFor="title"&gt;Skill name:&lt;/label&gt;
            &lt;input
              className={formik.errors.title &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >"error"}</span>
              name="title"
              placeholder="Enter skill name"
              value={formik.values.title}
              onChange={formik.handleChange}
            /&gt;
            {formik.errors.title &amp;&amp; (
<span class="branch-1 cbranch-no" title="branch not covered" >              &lt;div className="errors"&gt;{formik.errors.title}&lt;/div&gt;</span>
            )}
          &lt;/div&gt;
&nbsp;
          &lt;div className="skills__inputs"&gt;
            &lt;label htmlFor="range"&gt;Skill range:&lt;/label&gt;
            &lt;input
              className={formik.errors.range &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >"error"}</span>
              name="range"
              placeholder="Enter skill range"
              value={formik.values.range}
              onChange={formik.handleChange}
            /&gt;
            {formik.errors.range &amp;&amp; (
<span class="branch-1 cbranch-no" title="branch not covered" >              &lt;div className="errors"&gt;{formik.errors.range}&lt;/div&gt;</span>
            )}
          &lt;/div&gt;
          &lt;Button
            type="submit"
            text="Add skill"
            disabled={!(formik.dirty &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >formik.isValid)</span>}
          /&gt;
        &lt;/form&gt;
      )}
      &lt;div className="skills__content"&gt;
        {skills?.map(<span class="fstat-no" title="function not covered" >(i</span>tem, idx) =&gt; {
<span class="cstat-no" title="statement not covered" >          return (</span>
            &lt;div
              key={idx}
              style={{ width: `${item.range}%` }}
              className="skills__content-value"&gt;
              &lt;p className="skills__content-value-title"&gt;{item.title}&lt;/p&gt;
            &lt;/div&gt;
          );
        })}
      &lt;/div&gt;
      &lt;div className="skills__progress"&gt;
        &lt;div className="skills__progress-value"&gt;&lt;/div&gt;
        &lt;div className="skills__progress-value"&gt;&lt;/div&gt;
      &lt;/div&gt;
      &lt;div className="skills__levels"&gt;
        &lt;div className="skills__levels-title"&gt;
          &lt;p&gt;Beginner&lt;/p&gt;
          &lt;p&gt;Proficient&lt;/p&gt;
        &lt;/div&gt;
        &lt;div className="skills__levels-title"&gt;
          &lt;p&gt;Expert&lt;/p&gt;
          &lt;p&gt;Master&lt;/p&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  );
};
&nbsp;
export default Skills;
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Thu Oct 07 2021 11:52:08 GMT-0700 (Pacific Daylight Time)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    