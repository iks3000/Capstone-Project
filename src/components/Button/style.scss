@import "../../assets/styles/variables.scss";
%buttonProps {
    font: 400 16px/19px $primary-fonts;
    border-radius: 5px;
    padding: 10px 20px;
    cursor: pointer;
    border: none;
    transition: all 0.3s ease;
}

.button {
    @extend %buttonProps;
    background-color: $oxfordBlue;
    color: $white;
    &:hover {
        background-color: $green;
    }
}

.button-disabled {
    @extend %buttonProps;
    background-color: $raven;
    color: $midnight;
    opacity: 0.7;
    cursor: not-allowed;
}