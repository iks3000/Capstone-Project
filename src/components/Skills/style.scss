@import "../../assets/styles/_variables.scss";
@import "../../assets/styles/_extends.scss";
.skills {
    margin: 50px 0;
    &__header33 {
        display: flex;
        justify-content: space-between;
        align-items: baseline;
    }

    &__title {
        @extend %title;
    }

    &__button-ico {
        margin-right: 10px;
    }

    &__form {
        border: 1px solid $green;
        padding: 20px 15px;
        border-radius: 5px;
    }

    &__inputs {
        @extend %content-text;
        position: relative;
        margin-bottom: 22px;

        label {
            margin-right: 20px;
        }

        input {
            width: 270px;
            height: 40px;
            border: 1px solid $raven;
            border-radius: 5px;
            padding: 10px;
            outline: none;
        }

        input.error {
            border: 1px solid $red;
        }

        .errors {
            font-size: 12px;
            position: absolute;
            color: $red;
        }
    }

    &__content {
        width: 100%;
        margin: 35px 0 15px 0;
    }

    &__content-value {
        background-color: $green;
        margin-top: 5px;
        padding: 4px 15px;
    }

    &__content-value-title {
        color: $white;
        min-width: 20px;
        @extend %content-text;
    }

    &__progress {
        width: 100%;
        height: 10px;
        margin-top: 24px;
        display: flex;
        justify-content: space-between;
        border-bottom: 1px solid $raven;
    }

    &__progress-value {
        width: 20%;
        height: 10px;
        border-style: solid;
        border-width: 0 1px;
        border-color: $raven;
    }

    &__levels {
        width: 100%;
        display: flex;
        justify-content: space-between;
    }

    &__levels-title {
        width: 22%;
        margin-top: 10px;
        display: flex;
        justify-content: space-between;

        p {
            color: $raven;
            font-size: 12px;
            @extend %content-text;
        }
    }

}
