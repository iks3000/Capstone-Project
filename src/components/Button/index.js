import "./style.scss";

const Button = ({ icon, text = "Click", onClick, disabled, type }) => {
  return (
    <button
      className={disabled ? "button-disabled" : "button"}
      onClick={onClick}
      disabled={disabled}
      type={type}
    >
      {icon}
      <span>{text}</span>
    </button>
  );
};

export default Button;
