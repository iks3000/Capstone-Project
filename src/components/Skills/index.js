import { useEffect, useState } from "react";
import Button from "../Button";
import { useFormik } from "formik";
import * as Yup from "yup";
import { useDispatch } from "react-redux";
import { postSkills } from "../../app/features/skills/skillsSlice";
import PenLine from '../../assets/images/icons/penLine.svg';
import "./style.scss";


const Skills = () => {
  const [isOpen, setIsOpen] = useState(false);
  const [skills, setSkills] = useState(() => {
    const savedSkills = localStorage.getItem("skills");
    const initialValue = JSON.parse(savedSkills);
    return initialValue || [];
  });

  const dispatch = useDispatch();

  //validation schema for skill name and range
  const SkillsSchema = Yup.object().shape({
    title: Yup.string().required("Skill title is a required field"),
    range: Yup.number()
      .min(10, "Skill range must be greater than or equal to 10")
      .max(100, "Skill range must be less than or equal to 100")
      .required("Skill range is a required field"),
  });

  const formik = useFormik({
    initialValues: {
      title: "",
      range: "",
    },
    validationSchema: SkillsSchema,
    onSubmit: (values, { resetForm }) => {
      setSkills((prev) => [...prev, values]);
      dispatch(postSkills(values));
      resetForm();
    },
  });

  useEffect(() => {
    //saving skills to local storage
    localStorage.setItem("skills", JSON.stringify(skills));
  }, [skills]);

  const handleOpenEdit = () => {
    setIsOpen(!isOpen);
  };

  return (
    <div className="skills" id="skillsComponent">

      <div className="skills__header33">
        <h2 className="skills__title">Skills</h2>
        <Button
          icon={<img className="skills__button-ico" src={PenLine} alt="Pen with Line" />}
          text="Open edit"
          onClick={handleOpenEdit}
        />
      </div>
      {isOpen && (
        <form className="skills__form" onSubmit={formik.handleSubmit}>
          <div className="skills__inputs">
            <label htmlFor="title">Skill name:</label>
            <input
              className={formik.errors.title && "error"}
              name="title"
              placeholder="Enter skill name"
              value={formik.values.title}
              onChange={formik.handleChange}
            />
            {formik.errors.title && (
              <div className="errors">{formik.errors.title}</div>
            )}
          </div>

          <div className="skills__inputs">
            <label htmlFor="range">Skill range:</label>
            <input
              className={formik.errors.range && "error"}
              name="range"
              placeholder="Enter skill range"
              value={formik.values.range}
              onChange={formik.handleChange}
            />
            {formik.errors.range && (
              <div className="errors">{formik.errors.range}</div>
            )}
          </div>
          <Button
            type="submit"
            text="Add skill"
            disabled={!(formik.dirty && formik.isValid)}
          />
        </form>
      )}
      <div className="skills__content">
        {skills?.map((item, idx) => {
          return (
            <div
              key={idx}
              style={{ width: `${item.range}%` }}
              className="skills__content-value">
              <p className="skills__content-value-title">{item.title}</p>
            </div>
          );
        })}
      </div>
      <div className="skills__progress">
        <div className="skills__progress-value"></div>
        <div className="skills__progress-value"></div>
      </div>
      <div className="skills__levels">
        <div className="skills__levels-title">
          <p>Beginner</p>
          <p>Proficient</p>
        </div>
        <div className="skills__levels-title">
          <p>Expert</p>
          <p>Master</p>
        </div>
      </div>
    </div>
  );
};

export default Skills;
