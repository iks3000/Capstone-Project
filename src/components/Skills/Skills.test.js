import React from "react";
import { shallow } from "enzyme";
import Skills from "./index";
import Button from "../Button/index";

it("should render Skills component", () => {
    expect(shallow(<Skills />)).toMatchSnapshot();
});

const skills = [{
    id: 1,
    name: "React",
    range: 80,
}, ];

test("we should have skills", () => {
    expect(skills).toEqual(
        expect.arrayContaining([
            expect.objectContaining({ id: 1, name: "React", range: 80 }),
        ])
    );
});

describe("<TestComponent />", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow( <Skills /> );
    });

    it("has the initial state isOpen false", () => {
        expect(wrapper.state()).toEqual({ isOpen: false });
    });

    describe("should open skills", () => {
        it("setting iOpen to true on click", () => {
            wrapper.find(Button).props().onClick();
            expect(wrapper.state()).toEqual({ isOpen: true });
        });
    });


    describe("should close skills", () => {
        it("setting isOpen to false on click", () => {
            wrapper.find(Button).props().onClick();
            expect(wrapper.state()).toEqual({ isOpen: false });
        });
    });
});

