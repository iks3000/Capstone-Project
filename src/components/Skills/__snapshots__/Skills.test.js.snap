// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Skills component 1`] = `
<div
  className="skills"
  id="skillsComponent"
>
  <div
    className="skills__header33"
  >
    <h2
      className="skills__title"
    >
      Skills
    </h2>
    <Button
      icon={
        <img
          alt="Pen with Line"
          className="skills__button-ico"
          src="penLine.svg"
        />
      }
      onClick={[Function]}
      text="Open edit"
    />
  </div>
  <div
    className="skills__content"
  />
  <div
    className="skills__progress"
  >
    <div
      className="skills__progress-value"
    />
    <div
      className="skills__progress-value"
    />
  </div>
  <div
    className="skills__levels"
  >
    <div
      className="skills__levels-title"
    >
      <p>
        Beginner
      </p>
      <p>
        Proficient
      </p>
    </div>
    <div
      className="skills__levels-title"
    >
      <p>
        Expert
      </p>
      <p>
        Master
      </p>
    </div>
  </div>
</div>
`;
